src/coreclr/CMakeLists.txt

```diff
@@ -125,13 +131,13 @@ include(clrdefinitions.cmake)
 
 if(FEATURE_STANDALONE_GC)
   add_definitions(-DFEATURE_STANDALONE_GC)
-  if (CLR_CMAKE_BUILD_SUBSET_RUNTIME)
-    add_subdirectory(gc)
-  endif (CLR_CMAKE_BUILD_SUBSET_RUNTIME)
+  add_subdirectory(gc)
 endif(FEATURE_STANDALONE_GC)
 
-if(CLR_CMAKE_BUILD_SUBSET_RUNTIME AND NOT CLR_CROSS_COMPONENTS_BUILD)
+if(NOT CLR_CROSS_COMPONENTS_BUILD)
+  set(CLR_SINGLE_FILE_HOST_ONLY 1)
   add_subdirectory(${CLR_SRC_NATIVE_DIR}/corehost/apphost/static Corehost.Static)
+  add_dependencies(runtime singlefilehost)
```

Comment:
We want to build the "singlefilehost" as part of the "runtime" component. Otherwise we'd build it either as its own component (which no one will remember to specify) or as part of the fallback component (which would make all components build it, which is undesirable).

Reply:
This seems fine for now. Eventually I'd prefer we specify dependencies on the top level projects and people use subsets less. That's something we can address when we're ready.